### ISSUE TITLE
Add MeshRouter Customization to config.json

### ISSUE THREAD

**Issue by vitko-bg** [Ylianst/MeshCentral#3688]
Just realised that almost all customization is server based exclude MeshCentralRouter. All files (text, logo etc) have to be downloaded and located to MeshCentralRouter.exe location.
In my practice I downloading MeshRouter directly from the MC server and have to have prepared customization files on the target PC.

Can MeshCentralRouter customization be added to config.json please, so it can look like this:

  "_agentCustomization": {
         "displayName": "Compagny® Product™",
         "description": "Compagny® Product™ agent for remote monitoring, management and assistance.",
         "companyName": "Compagny",
         "serviceName": "compagnyagent",
         "fileName": "compagnyagent"
       },
       "_assistantCustomization": {
         "title": "Compagny® Product™",
         "image": "assistant-logo.png",
         "fileName": "compagny"
       },
       "_androidCustomization": {
         "title": "Compagny® Product™",
         "subtitle": "Product Subtitle™",
         "image": "assistant-logo.png"
       },
       "_routerCustomization": {
         "title": "Compagny® Product™",
         "subtitle": "Product Subtitle™",
         "image": "router-logo.png"
**OR**
        **"path-to-customization-folder": "/meshcentral/meshcentral-data/routercustomization"
with all files inside**
      },

In short to implement https://github.com/Ylianst/MeshCentralRouter/issues/10#issuecomment-784460586 to MC server

Thanks.

**Comment by Ylianst** [comment]
It's possible but lots of work, not sure when I will get to this.

**Comment by MatinatorX** [comment]
You might try creating an NSIS installer that includes all your additional files. We did it for this exact reason and to bundle in some other utilities we use. You can even add the registry entries during install that set the server name and options like this:

```
  !insertmacro WriteRegStringIfUndef HKCR "mcrouter" "" "MeshCentral Router"
  !insertmacro WriteRegStringIfUndef HKCR "mcrouter" "URL Protocol" ""
  !insertmacro WriteRegStringIfUndef HKCR "mcrouter\shell" "" ""
  !insertmacro WriteRegStringIfUndef HKCR "mcrouter\shell\open" "" ""
  !insertmacro WriteRegStringIfUndef HKCR "mcrouter\shell\open\command" "" "$\"$INSTDIR\meshcentralrouter.exe$\" $\"%1$\"" 
  !insertmacro WriteRegStringIfUndef HKCU "SOFTWARE\Open Source\MeshCentral Router" "" ""
  !insertmacro WriteRegStringIfUndef HKCU "SOFTWARE\Open Source\MeshCentral Router" "ServerName" "my.domain.com"
  !insertmacro WriteRegStringIfUndef HKCU "SOFTWARE\Open Source\MeshCentral Router" "Show Group Names" "0"
  !insertmacro WriteRegStringIfUndef HKCU "SOFTWARE\Open Source\MeshCentral Router" "Show Offline Devices" "0"
  !insertmacro WriteRegStringIfUndef HKCU "SOFTWARE\Open Source\MeshCentral Router" "DevDoubleClickClickAction" "0"
```

Once you have an installer made, there's a few ways you can make it available for download by your users. I stuck ours in the `meshcentral-web/public` directory, then added a nice big download link to the `"Footer":` section in config.json, since you can use html there.

@Ylianst might know a better way to make a download link or perhaps even how to replace the router executable that gets downloaded normally.

**Comment by vitko-bg** [comment]
Any update on the above please?
Thanks.
